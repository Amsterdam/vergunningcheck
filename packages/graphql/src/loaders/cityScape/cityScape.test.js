require("@spotify/polly-jest-presets");
const { loaderFunctionsTest } = require("../../testUtils");
const cityScape = require(".");
const mock = require("./__mocks__/1477.json");

const key1 = "f0ec3ea9-2ce4-4022-aba2-ab63ec448db9";
const key2 = "1477";

describe("CityScape loader", () => {
  describe("Basic functions", loaderFunctionsTest(cityScape, key1, key2));

  test("reducer", async () => {
    expect(cityScape.reducer(mock)).toMatchSnapshot();
  });

  test("findAddress", async () => {
    const x = await cityScape.load(["1477"]);
    expect(await x[0]).toMatchSnapshot();
  });

  test("findAddress", async () => {
    const x = await cityScape.load(["f0ec3ea9-2ce4-4022-aba2-ab63ec448db9"]);
    expect(await x[0]).toMatchSnapshot();
  });

  test("findAddress", async () => {
    const x = await cityScape.load(["0"]);
    expect(await x[0]).toMatchSnapshot();
  });
});
