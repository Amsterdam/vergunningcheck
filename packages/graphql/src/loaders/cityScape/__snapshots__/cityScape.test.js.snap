// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityScape loader Basic functions module result 1`] = `
Array [
  Object {
    "scope": "MUNICIPAL",
  },
]
`;

exports[`CityScape loader Basic functions module result 2`] = `
Array [
  Object {
    "scope": "NATIONAL",
  },
]
`;

exports[`CityScape loader findAddress 1`] = `
Object {
  "scope": "NATIONAL",
}
`;

exports[`CityScape loader findAddress 2`] = `
Object {
  "scope": "MUNICIPAL",
}
`;

exports[`CityScape loader findAddress 3`] = `
Object {
  "scope": undefined,
}
`;

exports[`CityScape loader reducer 1`] = `
Object {
  "scope": "NATIONAL",
}
`;
